chart/compass/charts/instance-creator/templates/deployment.yaml

```diff
@@ -0,0 +1,120 @@
+              containerPort: {{ .Values.deployment.args.containerPort }}
+              protocol: TCP
+          resources:
+            {{- toYaml .Values.deployment.resources | nindent 12 }}
+          {{- with .Values.deployment.securityContext }}
+          securityContext:
+{{ toYaml . | indent 12 }}
+          {{- end }}
+          env:
+            - name: APP_API_ROOT_PATH
+              value: {{ .Values.global.instanceCreator.apiRootPath }}
+            - name: APP_ADDRESS
+              value: "0.0.0.0:{{ .Values.deployment.args.containerPort }}"
+            - name: APP_HTTP_CLIENT_SKIP_SSL_VALIDATION
+              value: {{ $.Values.global.http.client.skipSSLValidation | quote }}
+            - name: APP_JWKS_ENDPOINT
+              value: {{ .Values.global.instanceCreator.authentication.jwksEndpoint | quote }}
+            - name: APP_SM_CLIENT_TIMEOUT
+              value: {{ .Values.global.instanceCreator.client.timeout }}
+            - name: APP_TENANT_INFO_ENDPOINT
+              value: https://{{ .Values.global.gateway.tls.host }}.{{ .Values.global.ingress.domainName }}{{ .Values.global.director.info.path }}
+            - name: APP_TENANT_INFO_REQUEST_TIMEOUT
+              value: {{ .Values.global.instanceCreator.tenantInfo.requestTimeout }}
+            - name: APP_TENANT_INFO_INSECURE_SKIP_VERIFY
+              value: {{ $.Values.global.http.client.skipSSLValidation | quote }}
+            - name: APP_SM_INSTANCES_SECRET_PATH
+              value: "{{ .Values.global.instanceCreator.secrets.path }}/{{ .Values.global.instanceCreator.secrets.key }}"
+            - name: APP_SM_INSTANCE_CLIENT_ID_PATH
+              value: {{ .Values.global.instanceCreator.clientIdPath }}
+            - name: APP_SM_INSTANCE_SM_URL_PATH
+              value: {{ .Values.global.instanceCreator.smUrlPath }}
+            - name: APP_SM_INSTANCE_TOKEN_URL_PATH
+              value: {{ .Values.global.instanceCreator.tokenURLPath }}
+            - name: APP_SM_INSTANCE_APP_NAME_PATH
+              value: {{ .Values.global.instanceCreator.appNamePath }}
+            - name: APP_SM_INSTANCE_CERTIFICATE_PATH
+              value: {{ .Values.global.instanceCreator.certificatePath }}
+            - name: APP_SM_INSTANCE_CERTIFICATE_KEY_PATH
+              value: {{ .Values.global.instanceCreator.certificateKeyPath }}
+            - name: APP_EXTERNAL_CLIENT_CERT_SECRET_NAME
+              value: "{{ .Values.global.externalCertConfiguration.secrets.externalClientCertSecret.name }}"
+            - name: APP_SM_INSTANCE_OAUTH_TOKEN_PATH
+              value: {{ .Values.global.instanceCreator.oauthTokenPath }}
+            - name: APP_SM_ASYNC_API_TICKER
+              value: {{ .Values.global.instanceCreator.ticker }}
+            - name: APP_SM_ASYNC_API_TIMEOUT
+              value: {{ .Values.global.instanceCreator.timeout }}
+          livenessProbe:
+            httpGet:
+              port: {{.Values.deployment.args.containerPort }}
```

Comment:
Could we use { .Values.global.instanceCreator.port }}? Same for the readinessProbe

Saw it above the code again, maybe the `{{.Values.deployment.args.containerPort }}` is not needed

Reply:
We could but from my PoV it is a bit safer to use the 'local' chart `values.yaml` for these kind of deployment related values because it's scoped only for the deployment. And for the `global` value, as it can be used in more places (since it's global), if it changes in the future for whatever reason the deployment will be affected.